core/types/types.go

```diff
@@ -44,6 +44,13 @@ type Validator struct {
 	Power  int64    `json:"power"`
 }
 
+// ByPubKey implements sort.Interface based on the PubKey field.
+type ByPubKey []Validator
+
+func (a ByPubKey) Len() int           { return len(a) }
+func (a ByPubKey) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByPubKey) Less(i, j int) bool { return a[i].PubKey.String() < a[j].PubKey.String() }
```

Comment:
Sorting in Go has evolved a lot.  Back in the day you'd see a ton of `sort.Interface` implementations, but they realized the Len and Swap were almost always identical boilerplate.  Enter `sort.Slice` where you specify just the less function.  But now we would use the modern `slices.SortFunc` where `bytes.Compare` satisfies the new comparison (returning an `int` rather than a less bool).